### PR TITLE
Streamlink Dailymotion source

### PR DESCRIPTION
**Pull request by Dum4G**
Dailymotion is tokenized source so yes, it's tokens do change (wow)
To make things worse it returns 403 on expired tokens
Dailymotion themselves are unhappy if you try to avoid their own player
https://stackoverflow.com/a/54980745

The only link that wasn't from dailymotion in Kenya playlist now returns 404

### CODE REVIEW

**Comment by Dum4G** [comment]
@brianmusundi do you know any other reliable sources for Kenian playlist?

File: channels/ke.m3u [hunk 1]
```diff
@@ -1,17 +1,5 @@
...
-#EXTINF:-1 tvg-id="K24 Kenya" tvg-name="K24" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/K24-1.jpg" group-title="",K24 Kenya
-https://stream-03.ix7.dailymotion.com/sec(uK1M-ag2Z_MzR7ZEmmKyDRjsqRv2lKOg6XrvFYtoI4M)/dm/3/x6lvncs/d/live-3.m3u8
-#EXTINF:-1 tvg-id="Kameme TV Kenya" tvg-name="Kameme TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/ke-kameme-tv-1383.jpg" group-title="",Kameme TV Kenya
-https://stream-09.dc3.dailymotion.com/sec(nitAgzxq_qCa0CId9XzFH7jTjfrllz_yrqsH26iNPGw)/dm/3/x6ol8sj/d/live-3.m3u8
-#EXTINF:-1 tvg-id="Kass TV Kenya" tvg-name="Kass TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/kasstvlogo.png" group-title="",Kass TV Kenya
-https://goliveafrica.media:9998/live/60755313b36db/index.m3u8
-#EXTINF:-1 tvg-id="KBC Kenya" tvg-name="KBC" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/KBC-1.jpg" group-title="",KBC Kenya
-https://stream-01.dc3.dailymotion.com/sec(86G48EQSWKUUFPhHXEV5xmRLVrEbPYAHOZrsRqUifEc)/dm/3/x74211t/s/live-2.m3u8
```
Comment (freearhey) [inline channels/ke.m3u:11]: Working link.
image

File: channels/ke.m3u [hunk 2]
```diff
@@ -1,17 +1,5 @@
...
-#EXTINF:-1 tvg-id="KTN Kenya" tvg-name="KTN Kenya" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/KTN-1.jpg" group-title="",KTN Kenya
-https://stream-01.dc3.dailymotion.com/sec(URBhp0zYv0oL3J-5svQBhvz9QRZJxxADFoCl9bPTiMY)/dm/3/x7l3lxv/s/live-3.m3u8
-#EXTINF:-1 tvg-id="NTV Kenya" tvg-name="NTV Kenya" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/NTV-1.jpg" group-title="",NTV Kenya
-https://stream-07.ix7.dailymotion.com/sec(oqY34t1NIwbiLJBkRzm88jOGQkwE9lxm_pkfjAcjmo8)/dm/3/x6shkab/d/live-3.m3u8
```
Comment (freearhey) [inline channels/ke.m3u:15]: image

**Review by freearhey — CHANGES_REQUESTED** [review]
Please check the marked links again.

**Comment by Dum4G** [comment]
@brianmusundi please check if those links are dead or just geoblocked


```
#EXTINF:-1 tvg-logo="https://cdn.nation.co.ke/assets/ntv/kenya/ntv-ke-logo.png" group-title="Kenya",NTV Kenya
http://ms-azure-video.gslb.startimestv.com/live/NTV_928_500_480x360_195092/playlist.m3u8
#EXTINF:-1 tvg-logo="https://www.standardmedia.co.ke/ktnhome/images/logo.png" group-title="Kenya",KTN Home
http://ms-azure-video.gslb.startimestv.com/live/KTN_HOME_916_500_480x360_195036/playlist.m3u8
#EXTINF:-1 tvg-logo="https://img.scooper.news/v2-EagleNews/Eagle-NewsImage/DetailImage/RealNews/20180924/a8b49753e599314503f454fa5a3dd802.png" group-title="Kenya",KTN Burudani
https://1655721754.rsc.cdn77.org/1655721754/tracks-v1a1/mono.m3u8
#EXTINF:-1 tvg-logo="https://www.farmers.co.ke/assets/images/logo.png" group-title="Kenya",KTN Farmer
https://1933667380.rsc.cdn77.org/1933667380/tracks-v1a1/mono.m3u8
#EXTINF:-1 tvg-logo="https://www.k24tv.co.ke/wp-content/uploads/2019/09/k24-1.png" group-title="Kenya",K24
http://ms-azure-video.gslb.startimestv.com/live/K24_892_500_480x360_194014/playlist.m3u8
#EXTINF:-1 tvg-logo="http://inoorotv.co.ke/wp-content/themes/inoorotv/images/inoorotv_logo.png" group-title="Kenya",Inooro TV
https://vidhttps.r.worldssl.net/inoorotv-origin/liveabr/inoorotv-origin/live3/chunks.m3u8
#EXTINF:-1 tvg-logo="http://kasstv.co.ke/wp-content/uploads/2018/01/cropped-kasstv.png" group-title="Kenya",Kass TV
http://197.232.56.13:1935/live/_definst_/kasstv/playlist.m3u8

```

**Comment by brianmusundi** [comment]
> http://ms-azure-video.gslb.startimestv.com/live/KTN_HOME_916_500_480x360_195036/playlist.m3u8

They aren't. They just don't work

**Comment by freearhey** [comment]
Links updated via #2829.

**Comment by Dum4G** [comment]
@brianmusundi we use group-title tag to define a category (Kids, News, Shopping etc)

**Comment by brianmusundi** [comment]
> @brianmusundi we use group-title tag to define a category (Kids, News, Shopping etc)

I intended to do that at first but I was copying info from my personal list. It's harder to find local channels because they'll mix with the US local channels. That's why I gave the group name as Kenyan even though they should have been local.  Also our local channels tend to have a lot of mixed content. They have news, movies, music and kid shows all at different times so it doesn't necessarily confine to one group.

**Comment by freearhey** [comment]
@Dum4G For others to see your proposal, it is better to publish it [here](https://github.com/iptv-org/iptv/issues/new?assignees=&labels=discussion&template=----feature-request.md&title=Propose%3A+xxx).

**Comment by Dum4G** [comment]
So far we have only 1 contributor caring for DailyMotion streams and the only country that relies heavily on the source. Unless we find a way to automatically revoke those links it's an exceptional case

**Comment by Sphinxroot** [comment]
#2781

File: channels/ke.m3u [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
+https://tvsn-i.akamaihd.net/hls/live/261837/tvsn/tvsn_750.m3u8
 #EXTINF:-1 tvg-id="Ebru TV Kenya" tvg-name="Ebru TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/download-1.jpg" group-title="",Ebru TV Kenya
 https://stream-01.dc3.dailymotion.com/sec(yMMZhXcH3F9vUXdVEAQnDLr2SM0UWwYkuLICgaGcRKY)/dm/3/x67n3k1/s/live-2.m3u8
-#EXTINF:-1 tvg-id="Inooro TV Kenya" tvg-name="Inooro TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/InooroTV-1.jpg" group-title="",Inooro TV Kenya
+#EXTINF:-1 tvg-id="Inooro TV Kenya" tvg-name="Inooro TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/InooroTV-1.jpg" group-title="",Inooro TV Kenya [Expiring Token]
 https://stream-01.ix7.dailymotion.com/sec(IbeXB-fEZXft8Mu82uFOZQ89_KtucjBJ-iNVupUhcks)/dm/3/x7ttbvq/s/live-4.m3u8
+#EXTINF:-1 tvg-id="K24 Kenya" tvg-name="K24" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/K24-1.jpg" tvg-url="" group-title="",K24 Kenya
+https://5f4db0f94b000.streamlock.net/k24/myStream/chunklist_w333466763.m3u8
```
Comment (freearhey) [inline channels/ke.m3u:9]: image

File: channels/ke.m3u [hunk 4]
```diff
@@ -1,17 +1,21 @@
 #EXTM3U
+#EXTINF:-1 tvg-id="Ebru TV Kenya" tvg-name="Ebru TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/download-1.jpg" tvg-url="" group-title="",Ebru TV Kenya
+https://tvsn-i.akamaihd.net/hls/live/261837/tvsn/tvsn_750.m3u8
 #EXTINF:-1 tvg-id="Ebru TV Kenya" tvg-name="Ebru TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/download-1.jpg" group-title="",Ebru TV Kenya
 https://stream-01.dc3.dailymotion.com/sec(yMMZhXcH3F9vUXdVEAQnDLr2SM0UWwYkuLICgaGcRKY)/dm/3/x67n3k1/s/live-2.m3u8
-#EXTINF:-1 tvg-id="Inooro TV Kenya" tvg-name="Inooro TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/InooroTV-1.jpg" group-title="",Inooro TV Kenya
+#EXTINF:-1 tvg-id="Inooro TV Kenya" tvg-name="Inooro TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/InooroTV-1.jpg" group-title="",Inooro TV Kenya [Expiring Token]
```
Comment (freearhey) [inline channels/ke.m3u:6]: Unsupported label

File: channels/ke.m3u [hunk 5]
```diff
@@ -1,17 +1,21 @@
...
 https://stream-01.ix7.dailymotion.com/sec(IbeXB-fEZXft8Mu82uFOZQ89_KtucjBJ-iNVupUhcks)/dm/3/x7ttbvq/s/live-4.m3u8
+#EXTINF:-1 tvg-id="K24 Kenya" tvg-name="K24" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/K24-1.jpg" tvg-url="" group-title="",K24 Kenya
+https://5f4db0f94b000.streamlock.net/k24/myStream/chunklist_w333466763.m3u8
 #EXTINF:-1 tvg-id="K24 Kenya" tvg-name="K24" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/K24-1.jpg" group-title="",K24 Kenya
 https://stream-06.ix7.dailymotion.com/sec(uK1M-ag2Z_MzR7ZEmmKyDXI_wCqDQusqgardAY5XCr0)/dm/3/x6lvncs/d/live-3.m3u8
-#EXTINF:-1 tvg-id="Kameme TV Kenya" tvg-name="Kameme TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/ke-kameme-tv-1383.jpg" group-title="",Kameme TV Kenya
+#EXTINF:-1 tvg-id="Kameme TV Kenya" tvg-name="Kameme TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/ke-kameme-tv-1383.jpg" group-title="",Kameme TV Kenya [Expiring Token]
```
Comment (freearhey) [inline channels/ke.m3u:12]: Unsupported label

File: channels/ke.m3u [hunk 6]
```diff
@@ -1,17 +1,21 @@
...
 https://stream-05.dc3.dailymotion.com/sec(nitAgzxq_qCa0CId9XzFH2v-wTlR6kxRn5k_zLyiuaA)/dm/3/x6ol8sj/d/live-3.m3u8
-#EXTINF:-1 tvg-id="KBC Kenya" tvg-name="KBC" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/KBC-1.jpg" group-title="",KBC Kenya
+#EXTINF:-1 tvg-id="KBC Kenya" tvg-name="KBC" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/KBC-1.jpg" group-title="",KBC Kenya [Expiring Token]
```
Comment (freearhey) [inline channels/ke.m3u:14]: Unsupported label

File: channels/ke.m3u [hunk 7]
```diff
@@ -1,17 +1,21 @@
...
 https://stream-01.dc3.dailymotion.com/sec(86G48EQSWKUUFPhHXEV5xmRLVrEbPYAHOZrsRqUifEc)/dm/3/x74211t/s/live-2.m3u8
-#EXTINF:-1 tvg-id="KTN Kenya" tvg-name="KTN Kenya" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/KTN-1.jpg" group-title="",KTN Kenya
+#EXTINF:-1 tvg-id="KTN Kenya" tvg-name="KTN Kenya" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/KTN-1.jpg" group-title="",KTN Kenya [Expiring Token]
```
Comment (freearhey) [inline channels/ke.m3u:16]: Unsupported label

File: channels/ke.m3u [hunk 8]
```diff
@@ -1,17 +1,21 @@
...
 https://stream-03.ix7.dailymotion.com/sec(URBhp0zYv0oL3J-5svQBhjeniu5faLrIc4RGd15HqC8)/dm/3/x7l3lxv/s/live-3.m3u8
-#EXTINF:-1 tvg-id="NTV Kenya" tvg-name="NTV Kenya" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/NTV-1.jpg" group-title="",NTV Kenya
+#EXTINF:-1 tvg-id="NTV Kenya" tvg-name="NTV Kenya" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/NTV-1.jpg" group-title="",NTV Kenya [Expiring Token]
```
Comment (freearhey) [inline channels/ke.m3u:18]: Unsupported label

File: channels/ke.m3u [hunk 9]
```diff
@@ -1,17 +1,21 @@
...
 https://stream-07.ix7.dailymotion.com/sec(oqY34t1NIwbiLJBkRzm88jOGQkwE9lxm_pkfjAcjmo8)/dm/3/x6shkab/d/live-3.m3u8
-#EXTINF:-1 tvg-id="Switch TV Kenya" tvg-name="Switch TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/Switch-TV-1.jpg" group-title="",Switch TV Kenya
-https://stream-04.dc3.dailymotion.com/sec(XpCeeMNY3Z5K08ImOEpf60-VQCStnQg7oVr_joW5a7o)/dm/3/x7sxle3/d/live-4.m3u8
+#EXTINF:-1 tvg-id="Switch TV Kenya" tvg-name="Switch TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/Switch-TV-1.jpg" group-title="",Switch TV Kenya [Expiring Token]
```
Comment (freearhey) [inline channels/ke.m3u:20]: Unsupported label

**Review by freearhey — CHANGES_REQUESTED** [review]
Please check the marked links again.

**Comment by Dum4G** [comment]
Gonna stay a draft until https://github.com/iptv-org/iptv/issues/2833 gets resolved and I'll find out what happened to streamlock link

File: channels/ke.m3u (hunk 3 above)
Comment (Dum4G) [reply channels/ke.m3u:9]: image
It's back. Not 24/7?

File: channels/ke.m3u (hunk 7 above)
Comment (Dum4G) [reply channels/ke.m3u:16]: It's time to have it then. This particular channel is geoblocked and has to be updated manually

**Comment by LaneSh4d0w** [comment]
I see there's quite a bit of a mix between pull requests. The Russian name changes belongs to another PR you've made. Please make sure that this PR corresponds to what you're proposing.

**Comment by Dum4G** [comment]
I did. Only 2 needed files are changed

File: channels/pk.m3u [hunk 10]
```diff
@@ -33,7 +33,9 @@ https://dcunilive38-lh.akamaihd.net/i/dclive_1@692677/master.m3u8
...
 #EXTINF:-1 tvg-id="MinhajTV.pk" tvg-name="Minhaj TV" tvg-country="PK" tvg-language="Urdu" tvg-logo="https://i.imgur.com/m8cS74j.png" group-title="",Minhaj TV (720p)
 https://api.new.livestream.com/accounts/547271/events/4237509/live.m3u8
-#EXTINF:-1 tvg-id="PTV.pk" tvg-name="PTV" tvg-country="PK" tvg-language="" tvg-logo="" group-title="",PTV
-https://stream-02.sg1.dailymotion.com/sec(vb0ikD3rmZA2RvEN2UsfwOmroW41RqGm0Ji7lmGKzmM)/dm/3/x5cr6b9/d/live-1.m3u8
 #EXTINF:-1 tvg-id="ZindagiTV.pk" tvg-name="Zindagi TV" tvg-country="PK" tvg-language="Urdu" tvg-logo="https://i.imgur.com/RtHeRwp.png" group-title="",Zindagi TV (576p)
 https://5ad386ff92705.streamlock.net/live_transcoder/ngrp:zindagitv.stream_all/chunklist.m3u8
+#EXTINF:-1 tvg-id="PTV.pk" tvg-name="PTV" tvg-country="PK" tvg-language="" tvg-logo="" group-title="",PTV
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x5cr6b9
+#EXTINF:-1 tvg-id="PTV.pk" tvg-name="PTV" tvg-country="PK" tvg-language="" tvg-logo="" group-title="",PTV
+http://salvahe.co/iptv-query?url=https://dai.ly/x5cr6b9
```
Comment (freearhey) [inline channels/pk.m3u:41]: The link doesn't work.

File: channels/ke.m3u [hunk 11]
```diff
@@ -1,17 +1,49 @@
...
+https://goliveafrica.media:9998/live/60755313b36db/index.m3u8
+#EXTINF:-1 tvg-id="GBSTV.ke" tvg-name="GBS TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/GBS-TV-1.jpg" tvg-url="" group-title="General",GBS TV [Not 24/7]
+https://goliveafrica.media:9998/live/6045ccbac3484/index.m3u8
+#EXTINF:-1 tvg-id="UTV.ke" tvg-name="UTV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/UTV-1.jpg" tvg-url="" group-title="General",UTV [Not 24/7]
+https://goliveafrica.media:9998/live/6049f726546e1/index.m3u8
+#EXTINF:-1 tvg-id="TV47.ke" tvg-name="TV47" tvg-country="KE" tvg-language="English" tvg-logo="https://tv47.co.ke/wp-content/uploads/2020/10/tv-47-logo-600-e1617870103726.png" tvg-url="" group-title="General",TV47
+https://iptv--iptv.repl.co/streamlink?url=https://www.twitch.tv/tv47kenya/
+#EXTINF:-1 tvg-id="TV47.ke" tvg-name="TV47" tvg-country="KE" tvg-language="English" tvg-logo="https://tv47.co.ke/wp-content/uploads/2020/10/tv-47-logo-600-e1617870103726.png" tvg-url="" group-title="General",TV47
+http://salvahe.co/iptv-query?url=https://www.twitch.tv/tv47kenya/
```
Comment (freearhey) [inline channels/ke.m3u:49]: The link doesn't work.

File: channels/ke.m3u [hunk 12]
```diff
@@ -1,17 +1,49 @@
...
-#EXTINF:-1 tvg-id="NTV Kenya" tvg-name="NTV Kenya" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/NTV-1.jpg" group-title="",NTV Kenya
-https://stream-07.ix7.dailymotion.com/sec(oqY34t1NIwbiLJBkRzm88jOGQkwE9lxm_pkfjAcjmo8)/dm/3/x6shkab/d/live-3.m3u8
-#EXTINF:-1 tvg-id="Switch TV Kenya" tvg-name="Switch TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/Switch-TV-1.jpg" group-title="",Switch TV Kenya
-https://stream-04.dc3.dailymotion.com/sec(XpCeeMNY3Z5K08ImOEpf60-VQCStnQg7oVr_joW5a7o)/dm/3/x7sxle3/d/live-4.m3u8
+#EXTINF:-1 tvg-id="NTV.ke" tvg-name="NTV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/NTV-1.jpg" group-title="General",NTV
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x6shkab
+#EXTINF:-1 tvg-id="NTV.ke" tvg-name="NTV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/NTV-1.jpg" group-title="General",NTV
+http://salvahe.co/iptv-query?url=https://dai.ly/x6shkab
```
Comment (freearhey) [inline channels/ke.m3u:31]: The link doesn't work.

File: channels/ke.m3u [hunk 13]
```diff
@@ -1,17 +1,49 @@
...
+https://5f4db0f94b000.streamlock.net/k24/myStream/chunklist_w333466763.m3u8
+#EXTINF:-1 tvg-id="K24.ke" tvg-name="K24" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/K24-1.jpg" group-title="General",K24
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x6lvncs
+#EXTINF:-1 tvg-id="K24.ke" tvg-name="K24" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/K24-1.jpg" group-title="General",K24
+http://salvahe.co/iptv-query?url=https://dai.ly/x6lvncs
+#EXTINF:-1 tvg-id="KamemeTV.ke" tvg-name="Kameme TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/ke-kameme-tv-1383.jpg" group-title="General",Kameme TV
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x6ol8sj
+#EXTINF:-1 tvg-id="KamemeTV.ke" tvg-name="Kameme TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/ke-kameme-tv-1383.jpg" group-title="General",Kameme TV
+http://salvahe.co/iptv-query?url=https://dai.ly/x6ol8sj
```
Comment (freearhey) [inline channels/ke.m3u:21]: The link doesn't work.

File: channels/ke.m3u [hunk 14]
```diff
@@ -1,17 +1,49 @@
...
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x7ttbvq
+#EXTINF:-1 tvg-id="InooroTV.ke" tvg-name="Inooro TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/InooroTV-1.jpg" group-title="General",Inooro TV
+http://salvahe.co/iptv-query?url=https://dai.ly/x7ttbvq
+#EXTINF:-1 tvg-id="K24.ke" tvg-name="K24" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/K24-1.jpg" tvg-url="" group-title="General",K24 [Not 24/7]
+https://5f4db0f94b000.streamlock.net/k24/myStream/chunklist_w333466763.m3u8
+#EXTINF:-1 tvg-id="K24.ke" tvg-name="K24" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/K24-1.jpg" group-title="General",K24
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x6lvncs
+#EXTINF:-1 tvg-id="K24.ke" tvg-name="K24" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/K24-1.jpg" group-title="General",K24
+http://salvahe.co/iptv-query?url=https://dai.ly/x6lvncs
```
Comment (freearhey) [inline channels/ke.m3u:17]: The link doesn't work.

File: channels/ke.m3u [hunk 15]
```diff
@@ -1,17 +1,49 @@
...
+#EXTINF:-1 tvg-id="KamemeTV.ke" tvg-name="Kameme TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/ke-kameme-tv-1383.jpg" group-title="General",Kameme TV
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x6ol8sj
+#EXTINF:-1 tvg-id="KamemeTV.ke" tvg-name="Kameme TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/ke-kameme-tv-1383.jpg" group-title="General",Kameme TV
+http://salvahe.co/iptv-query?url=https://dai.ly/x6ol8sj
+#EXTINF:-1 tvg-id="KBC.ke" tvg-name="KBC" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/KBC-1.jpg" group-title="General",KBC
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x74211t
+#EXTINF:-1 tvg-id="KBC.ke" tvg-name="KBC" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/KBC-1.jpg" group-title="General",KBC
+http://salvahe.co/iptv-query?url=https://dai.ly/x74211t
```
Comment (freearhey) [inline channels/ke.m3u:25]: The link doesn't work.

File: channels/ke.m3u [hunk 16]
```diff
@@ -1,17 +1,49 @@
...
-https://stream-01.dc3.dailymotion.com/sec(86G48EQSWKUUFPhHXEV5xmRLVrEbPYAHOZrsRqUifEc)/dm/3/x74211t/s/live-2.m3u8
-#EXTINF:-1 tvg-id="KTN Kenya" tvg-name="KTN Kenya" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/KTN-1.jpg" group-title="",KTN Kenya
+#EXTINF:-1 tvg-id="EbruTV.ke" tvg-name="Ebru TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/download-1.jpg" tvg-url="" group-title="General",Ebru TV
+https://tvsn-i.akamaihd.net/hls/live/261837/tvsn/tvsn_750.m3u8
+#EXTINF:-1 tvg-id="EbruTV.ke" tvg-name="Ebru TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/download-1.jpg" group-title="General",Ebru TV
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x67n3k1
+#EXTINF:-1 tvg-id="EbruTV.ke" tvg-name="Ebru TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2020/11/download-1.jpg" group-title="General",Ebru TV
+http://salvahe.co/iptv-query?url=https://dai.ly/x67n3k1
```
Comment (freearhey) [inline channels/ke.m3u:7]: The link doesn't work.

File: channels/ke.m3u [hunk 17]
```diff
@@ -1,17 +1,49 @@
...
+#EXTINF:-1 tvg-id="InooroTV.ke" tvg-name="Inooro TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/InooroTV-1.jpg" group-title="General",Inooro TV
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x7ttbvq
+#EXTINF:-1 tvg-id="InooroTV.ke" tvg-name="Inooro TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/InooroTV-1.jpg" group-title="General",Inooro TV
+http://salvahe.co/iptv-query?url=https://dai.ly/x7ttbvq
```
Comment (freearhey) [inline channels/ke.m3u:11]: The link doesn't work.

File: channels/ke.m3u [hunk 18]
```diff
@@ -1,17 +1,49 @@
...
-https://stream-04.dc3.dailymotion.com/sec(XpCeeMNY3Z5K08ImOEpf60-VQCStnQg7oVr_joW5a7o)/dm/3/x7sxle3/d/live-4.m3u8
+#EXTINF:-1 tvg-id="NTV.ke" tvg-name="NTV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/NTV-1.jpg" group-title="General",NTV
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x6shkab
+#EXTINF:-1 tvg-id="NTV.ke" tvg-name="NTV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/NTV-1.jpg" group-title="General",NTV
+http://salvahe.co/iptv-query?url=https://dai.ly/x6shkab
+#EXTINF:-1 tvg-id="SwitchTV.ke" tvg-name="Switch TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/Switch-TV-1.jpg" group-title="General",Switch TV
+https://iptv--iptv.repl.co/streamlink?url=https://dai.ly/x7sxle3
+#EXTINF:-1 tvg-id="SwitchTV.ke" tvg-name="Switch TV" tvg-country="KE" tvg-language="English" tvg-logo="https://kenyatv.tech/wp-content/uploads/2021/04/Switch-TV-1.jpg" group-title="General",Switch TV
+http://salvahe.co/iptv-query?url=https://dai.ly/x7sxle3
```
Comment (freearhey) [inline channels/ke.m3u:35]: The link doesn't work.

**Review by freearhey — CHANGES_REQUESTED** [review]
Please check the marked links again.

File: channels/ke.m3u (hunk 14 above)
Comment (Dum4G) [reply channels/ke.m3u:17]: image
Do I have to try the others?

Comment (LaneSh4d0w) [reply channels/ke.m3u:17]: The Salvahe links are dead.

**Review by freearhey — APPROVED** [review]

**Pull request merged**